pym/bob/scm/url.py

```diff
@@ -98,24 +96,49 @@ def asScript(self):
 cp -uT {URL} {FILE}
 """
         else:
+            # do command logic inside the shell script to leave it to the backend
             tpl = """
 {HEADER}
 mkdir -p {DIR}
 cd {DIR}
 if [ -e {FILE} ] ; then
-    curl {OPTIONS} -o {FILE} -z {FILE} {URL}
+    if [ -x "$(command -v curl)" ] ; then
+        OPTIONS="-sSgLf"
+        if [ {SKIPCERTCHECK} = True ] ; then
+            OPTIONS+="k"
+        fi
+        curl $OPTIONS -o {FILE} -z {FILE} {URL}
+    fi
 else
     (
         F=$(mktemp)
         trap 'rm -f $F' EXIT
         set -e
-        curl {OPTIONS} -o $F {URL}
+        if [ -x "$(command -v curl)" ] ; then
+            OPTIONS="-sSgLf"
+            if [ {SKIPCERTCHECK} = True ] ; then
+                OPTIONS+="k"
+            fi
+            curl $OPTIONS -o $F {URL}
+        elif [ -x "$(command -v wget)" ] ; then
+            OPTIONS="-q --no-glob"
+            if [ {SKIPCERTCHECK} = True ] ; then
+                OPTIONS+=" --no-check-certificate"
+            fi
+            wget $OPTIONS -O $F {URL}
+        else
+            >&2 echo "\nERROR: Don't know how to download. Please install 'curl'!\n"
+            # cleanup
+            rm -f $F
```

Comment:
This is unneeded. The `trap ...` at the start of the sub-shell already takes care of this.